05-JWT-Basics/preferred/middlewares/auth.js

```diff
@@ -0,0 +1,23 @@
+const { promisify } = require('util');
+const jwt = require('jsonwebtoken');
+
+const protect = async (req, res, next) => {
+  const token = req.headers?.authorization?.split(' ')[1];
```

Comment:
It's generally good practice to be very strict on what headers you're expecting, especially for security things like authentication/authorization. Commonly, with JWTs we ask users to pass an `Authorization` header that's prefixed with the text `Bearer`.

Currently your code accepts anything as long as there's a space followed by a valid JWT token. So it would also allow an `Authorization `token that looks like `"MySpecialPrefix <jwt token>"`. I would suggest updating the code to only specifically allow headers prefixed with `Bearer`